synapse/handlers/sync.py

```diff
@@ -1888,7 +1911,9 @@ async def handle_room_entries(room_entry: "RoomSyncResultBuilder") -> None:
         )
 
     async def _have_rooms_changed(
-        self, sync_result_builder: "SyncResultBuilder"
+        self,
+        sync_result_builder: "SyncResultBuilder",
+        un_partial_stated_rooms: Set[str],
```

Comment:
Consider annotating this argument as an `AbstractSet` (which means it will be immutable within the body of the function). See e.g. https://mypy-play.net/?mypy=latest&python=3.11&gist=1a927c3159d69fb2e368758fbd5f8159